src/WebSdk/Publish/Tasks/Tasks/MsDeploy/MSDeploy.cs

```diff
@@ -263,13 +264,24 @@ public string ExePath
 #else
                 if (string.IsNullOrEmpty(m_exePath))
                 {
-                    string programFilesX86 = System.Environment.ExpandEnvironmentVariables("%ProgramFiles(x86)%");
-                    string msdeployExePath = System.IO.Path.Combine("IIS", "Microsoft Web Deploy V3");
-                    m_exePath = IO.Path.Combine(programFilesX86, msdeployExePath);
+                    string programFiles = Environment.ExpandEnvironmentVariables("%ProgramFiles(x86)%");
+                    string msdeployExePath = IO.Path.Combine("IIS", "Microsoft Web Deploy V3");
```

Comment:
IIRC the directory name does not always include "V3". Perhaps that was with an older version that you don't support here.